server/graphql/v2/query/ExpensesQuery.ts

```diff
@@ -74,70 +74,72 @@ const updateFilterConditionsForReadyToPay = async (where, include): Promise<void
-      defaultValue: CHRONOLOGICAL_ORDER_INPUT_DEFAULT_VALUE,
-    },
-    minAmount: {
-      type: GraphQLInt,
-      description: 'Only return expenses where the amount is greater than or equal to this value (in cents)',
-    },
-    maxAmount: {
-      type: GraphQLInt,
-      description: 'Only return expenses where the amount is lower than or equal to this value (in cents)',
-    },
-    payoutMethodType: {
-      type: PayoutMethodType,
-      description: 'Only return expenses that use the given type as payout method',
-    },
-    dateFrom: {
-      type: GraphQLDateTime,
-      description: 'Only return expenses that were created after this date',
-    },
-    dateTo: {
-      type: GraphQLDateTime,
-      description: 'Only return expenses that were created after this date',
-    },
-    searchTerm: {
-      type: GraphQLString,
-      description: 'The term to search',
-    },
+const ExpenseQueryArgs = {
+  ...CollectionArgs,
+  fromAccount: {
+    type: AccountReferenceInput,
+    description: 'Reference of the account that submitted this expense',
+  },
+  account: {
+    type: AccountReferenceInput,
+    description: 'Reference of the account where this expense was submitted',
+  },
+  host: {
+    type: AccountReferenceInput,
+    description: 'Return expenses only for this host',
+  },
+  status: {
+    type: ExpenseStatusFilter,
+    description: 'Use this field to filter expenses on their statuses',
+  },
+  type: {
+    type: ExpenseType,
+    description: 'Use this field to filter expenses on their type (RECEIPT/INVOICE)',
+  },
+  tags: {
+    type: new GraphQLList(GraphQLString),
+    description: 'Only expenses that match these tags',
+    deprecationReason: '2020-06-30: Please use tag (singular)',
+  },
+  tag: {
+    type: new GraphQLList(GraphQLString),
+    description: 'Only expenses that match these tags',
+  },
+  orderBy: {
+    type: new GraphQLNonNull(ChronologicalOrderInput),
+    description: 'The order of results',
+    defaultValue: CHRONOLOGICAL_ORDER_INPUT_DEFAULT_VALUE,
+  },
+  minAmount: {
+    type: GraphQLInt,
+    description: 'Only return expenses where the amount is greater than or equal to this value (in cents)',
   },
-  async resolve(_: void, args, req: express.Request): Promise<CollectionReturnType> {
+  maxAmount: {
+    type: GraphQLInt,
+    description: 'Only return expenses where the amount is lower than or equal to this value (in cents)',
+  },
+  payoutMethodType: {
+    type: PayoutMethodType,
+    description: 'Only return expenses that use the given type as payout method',
+  },
+  dateFrom: {
+    type: GraphQLDateTime,
+    description: 'Only return expenses that were created after this date',
+  },
+  dateTo: {
+    type: GraphQLDateTime,
+    description: 'Only return expenses that were created after this date',
+  },
+  searchTerm: {
+    type: GraphQLString,
+    description: 'The term to search',
+  },
+};
+
+const ExpensesQuery: Query<typeof ExpenseQueryArgs> = {
```

Comment:
If `ExpenseQueryArgs` is an object, `typeof ExpenseQueryArgs` returns `object`. What's the point of typing with such a generic type?

Reply:
It is not that generic in this case. And after you get the `keyof` you'll be able to get all the argmuent keys you can have:
![Screenshot from 2021-08-11 09-43-15](https://user-images.githubusercontent.com/2119706/129030922-10d8dcb6-3f6a-
41f6-bada-df31781520c9.png)

Reply:
Just read about the following, I was originally reading it in a JavaScript way
https://www.typescriptlang.org/docs/handbook/2/typeof-types.html

Still looking like a hack, because only the keys are similar not the structure of the object itself. I'm no typescript expert.

How would we validate such argument? `fromAccount: {slug: "opensource"}`

Reply:
You're correct. This is just a naive solution to implement quick key-level arg validation without requiring manual typing of our objects.
The ideal solution would be integrating GraphQL and TS, which is quite more costly, both for refactoring existing code and for writing new code.